src/ebay-video/video.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { ComponentProps, FC, SyntheticEvent, useEffect, useRef, useState } from 'react'
+import classNames from 'classnames'
+import shaka from 'shaka-player/dist/shaka-player.ui'
+import 'shaka-player/dist/controls.css'
+
+import { filterByType } from '../common/component-utils'
+import { EbayIcon, EbayProgressSpinner } from '../index'
+import { VideoAction, VideoPlayView } from './types'
+import EbayVideoSource from './source'
+import { defaultVideoConfig, ERROR_ANOTHER_LOAD, ERROR_NO_PLAYER } from './const'
+import { customControls } from './controls'
+
+export type EbayVideoProps = ComponentProps<'video'> & {
+    width?: number;
+    height?: number;
+    thumbnail?: string;
+    action?: VideoAction;
+    volume?: number;
+    muted?: boolean;
+    volumeSlider?: boolean;
+    playView?: VideoPlayView;
+    cdnUrl?: string;
+    cssUrl?: string;
+    cdnVersion?: string;
+    a11yLoadText: string;
+    a11yPlayText: string;
+    errorText: string;
+    reportText: string;
+    onLoadError?: (err: any) => void;
+    onPlay?: (e: SyntheticEvent, { player: any }) => void;
+    onVolumeChange?: (e: SyntheticEvent, { volume: number, muted: boolean }) => void;
+    onReport?: () => void;
+};
+
+const EbayVideo: FC<EbayVideoProps> = ({
+    width,
+    height,
+    thumbnail,
+    action,
+    muted,
+    playView = 'inline',
+    a11yLoadText,
+    a11yPlayText,
+    reportText,
+    volumeSlider,
+    volume = 1,
+    errorText,
+    onVolumeChange = () => {
+    },
+    onLoadError = () => {
+    },
+    onPlay = () => {
+    },
+    onReport = () => {
+    },
+    children,
+    ...rest
+}) => {
+    const [loaded, setLoaded] = useState<boolean>()
+    const [buffering, setBuffering] = useState<boolean>()
+    const [playing, setPlaying] = useState<boolean>()
+    const [failed, setFailed] = useState<boolean>()
+
+    const containerRef = useRef<HTMLDivElement>(null)
+    const videoRef = useRef<HTMLVideoElement>(null)
+    const playerRef = useRef<any>(null)
+    const uiRef = useRef(null)
+
+    const sources = filterByType(children, EbayVideoSource).map(({ props }) => props)
+
+    const handleError = (err: any) => {
+        setLoaded(true)
+        setFailed(true)
+        onLoadError(err)
+    }
+
+    const loadSource = (index = 0) => {
+        if (!sources.length || !playerRef.current) return
+
+        setLoaded(false)
+        playerRef.current
+            .load(sources[index]?.src)
+            .then(() => {
+                setFailed(false)
+            })
+            .catch(err => {
+                console.error(err)
+                switch (err.code) {
+                    case ERROR_ANOTHER_LOAD:
+                        return
+                    case ERROR_NO_PLAYER:
+                        setTimeout(() => loadSource(index), 0)
```

Comment:
Any reason we are using setTimeout? Should we use `requestAnimateFrame` or `requestidlecallback`? We can add a comment here explaining why setTimeout